test/extensions/quic_listeners/quiche/envoy_quic_server_stream_test.cc

```diff
@@ -275,19 +277,28 @@ TEST_P(EnvoyQuicServerStreamTest, OutOfOrderTrailers) {
 
 TEST_P(EnvoyQuicServerStreamTest, ResetStreamByHCM) {
   receiveRequest(request_body_, false, request_body_.size() * 2);
+  if (quic::VersionUsesHttp3(quic_version_.transport_version)) {
+    EXPECT_CALL(quic_session_, MaybeSendStopSendingFrame(_, _));
+  }
+  EXPECT_CALL(quic_session_, MaybeSendRstStreamFrame(_, _, _));
   EXPECT_CALL(stream_callbacks_, onResetStream(_, _));
   quic_stream_->resetStream(Http::StreamResetReason::LocalReset);
   EXPECT_TRUE(quic_stream_->rst_sent());
 }
 
-TEST_P(EnvoyQuicServerStreamTest, EarlyResponseWithReset) {
+TEST_P(EnvoyQuicServerStreamTest, EarlyResponseWithStopSending) {
+  quic::SetVerbosityLogThreshold(1);
   receiveRequest(request_body_, false, request_body_.size() * 2);
   // Write response headers with FIN before finish receiving request.
   quic_stream_->encodeHeaders(response_headers_, true);
-  // Resetting the stream now means stop reading and sending QUIC_STREAM_NO_ERROR.
+  // Resetting the stream now means stop reading and sending QUIC_STREAM_NO_ERROR or STOP_SENDING.
+  if (quic::VersionUsesHttp3(quic_version_.transport_version)) {
+    EXPECT_CALL(quic_session_, MaybeSendStopSendingFrame(_, _));
+  } else {
+    EXPECT_CALL(quic_session_, MaybeSendRstStreamFrame(_, _, _));
+  }
```

Comment:
In other tests in this file, we always expect MaybeSendRstStreamFrame, and also expect a MaybeSendStopSendingFrame for h3.  This test expects only one of them to happen. Is it intended?

Reply:
Yes, in early response case, FIN is ready sent before resetting, so IETF quic will not send RST_STREAM again but only STOP_SENDING.

Reply:
Thanks for the explanation.